napari/_vispy/vendored/volume.py

```diff
@@ -947,6 +1072,45 @@ def relative_step_size(self, value):
         self._relative_step_size = value
         self.shared_program['u_relative_step_size'] = value
 
+    @property
+    def plane_position(self):
+        return self._plane_position
+
+    @plane_position.setter
+    def plane_position(self, value):
+        value = np.array(value, dtype=np.float32).ravel()
+        if value.shape != (3,):
+            raise ValueError('plane_position must be a 3 element array-like object')
+        self._plane_position = value
+        self.shared_program['u_plane_position'] = value[::-1]
+        self.update()
+
+    @property
+    def plane_normal(self):
```

Comment:
I'm unsure if this is in the style of the rest of the vispy code base, but we might consider making all of the plane attributes a single property. I could imagine like you did with the plane on our Image layer, I think it is nice to group the properties together.

Reply:
good idea! I'll try putting things in a simple dataclass on the vispy side